lib/instrument/headers.js

```diff
@@ -0,0 +1,260 @@
+    return {
+      magic: headerBuffer.readUInt32LE(0),
+      headerLength: headerBuffer.readUInt32LE(4),
+      fragmentId: headerBuffer.readUInt16LE(8),
+      fragmentCount: headerBuffer.readUInt16LE(10),
+      payloadLength: headerBuffer.readUInt32LE(12),
+      identifier: headerBuffer.readUInt32LE(16),
+      conversationIndex: headerBuffer.readUInt32LE(20),
+      channel: headerBuffer.readUInt32LE(24),
+      expectsReply: headerBuffer.readUInt32LE(28),
+    };
+  }
+}
+
+class DTXMessagePayloadHeader {
+  static build (data) {
+    const messageHeader = Buffer.alloc(DTX_MESSAGE_PAYLOAD_HEADER_LENGTH);
+    messageHeader.writeUInt32LE(data.flags, 0);
+    messageHeader.writeUInt32LE(data.auxLength, 4);
+    // eslint-disable-next-line no-undef
+    messageHeader.writeBigUInt64LE(BigInt(data.totalLength), 8);
+    return messageHeader;
+  }
+  static parse (headerBuffer) {
+    return {
+      flags: headerBuffer.readUInt32LE(0),
+      auxLength: headerBuffer.readUInt32LE(4),
+      totalLength: Number(headerBuffer.readBigInt64LE(8))
+    };
+  }
+}
+
+
+class DTXMessageAux {
+  constructor (data = []) {
+    this.data = data;
+  }
+  build () {
+    const messageAux = new DTXMessageAuxBuffer();
+    if (this.data.length === 0) {
+      return Buffer.alloc(0);
+    }
+    for (const arg of this.data) {
+      if (typeof arg === 'number') {
+        if (Math.abs(arg) < (2 ** 32)) {
+          messageAux.appendInt(arg);
+        } else if (Math.abs(arg) < (2 ** 64)) {
+          messageAux.appendLong(arg);
+        } else {
+          throw new Error('num too large');
+        }
+      } else {
+        messageAux.appendObject(arg);
+      }
+    }
+    return messageAux.getBytes();
+  }
+  static parse (headerBuffer) {
+    let cursor = 0;
+    // const magic = Number(headerBuffer.readBigInt64LE(0));
+    // cursor += 8;
+    const data = [];
+    const length = Number(headerBuffer.readBigInt64LE(8));
+    cursor += 16;
+    while (cursor <= length) {
+      const m = headerBuffer.readUInt32LE(cursor);
+      if (m !== 0xa) {
+        throw new Error('incorrect auxiliary magic');
+      }
+      const type = headerBuffer.readUInt32LE(cursor + 4);
+      cursor += 8;
+      switch (type) {
+        case 2: {
+          const strLen = headerBuffer.readUInt32LE(cursor);
+          cursor += 4;
+          const archived_data = headerBuffer.slice(cursor, cursor + strLen);
+          data.push(unarchive(archived_data));
+          cursor += strLen;
+          break;
+        }
+        case 3:
+          data.push(headerBuffer.readUInt32LE(cursor));
+          cursor += 4;
+          break;
+        case 4:
+          data.push(headerBuffer.readBigUInt64LE(cursor));
+          cursor += 8;
+          break;
+        case 6:
+          data.push(headerBuffer.readBigInt64LE(cursor));
+          cursor += 8;
+          break;
+      }
+    }
+    return data;
+  }
+}
+
+
+function InstrumentRPCParseError (data) {
```

Comment:
I would rather follow the common pattern here and make this function to a class with a constructor